Automattic/jetpack
# Boost: Fix CLS increase due to Critical CSS onload events

Fixes #20762
When different CSS files on a page loading at different speeds affect the same elements above the fold, adding Critical CSS increases the CLS score. This can be fixed by loading all the styles on the window `load` event.
Changes in this pull request contain the fix for enabling the window-load-stylesheet with the help of a filter.

#### Changes proposed in this Pull Request:

* Add `jetpack_boost_window_loaded_media` filter to differ the stylesheets on window `load` event.
* Note: The actual stylesheets loading happens with [onload_flip_stylesheets](https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/modules/critical-css/class-critical-css.php#L843)

#### Jetpack product discussion


UI options or automatic configuration can provide a better way to fix the issue. More discussion regarding the issue: pbIAo1-o2-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Go to a site with Jetpack Boost Plugin activated.
* Create and enable a mu-plugin as described here: #20762
* Navigate to wp-admin/admin.php?page=jetpack-boost.
* Enable `Optimize CSS Loading`
* Run a lighthouse report for the homepage.
    * CLS score should be greater than 0
* Filter `jetpack_boost_window_loaded_media` to `true` in a custom plugin
* Run a lighthouse report for the homepage.
    * CLS score should be reduced from the previous score.
